repositories/strategies/spacedOrderStrategy.js

```diff
@@ -1,104 +1,54 @@
-const { map, get } = require("lodash/fp");
+const { isEmpty, flow, last, clamp, getOr, isNil } = require("lodash/fp");
 
 const SPACING = 1000;
 
-const calculateMidPoint = (orderBefore, orderAfter) =>
-  Math.round((orderBefore + orderAfter) / 2);
+const calculateMidPoint = (a, b) => Math.round((a + b) / 2);
 
 const valuesHaveConverged = (orderBefore, orderAfter) =>
   Math.abs(orderAfter - orderBefore) < 2;
 
-const getNextOrderValue = (trx, SectionId) => {
+const makeSpaceAfter = (trx, SectionId, order) => {
   return trx("Pages")
-    .column(trx.raw(`COALESCE(max("order"), 0) + ${SPACING} as "order"`))
-    .where({ SectionId, isDeleted: false });
-};
-
-const setOrder = (trx, id, SectionId, order) => {
-  return trx("Pages")
-    .where({ id })
-    .update({ order, SectionId })
-    .returning("*");
+    .where({ SectionId })
+    .andWhere("order", ">", order)
+    .increment("order", SPACING);
 };
 
-const getAdjacentRows = (trx, id, sectionId, position) => {
-  return trx("Pages")
+const getPages = (trx, SectionId, id) => {
+  const query = trx("PagesView")
     .columns("id", "order")
-    .where({
-      SectionId: sectionId,
-      isDeleted: false
-    })
-    .where("id", "!=", id)
-    .orderBy("order")
-    .offset(Math.max(0, position - 1))
-    .limit(2)
-    .then(map("order"));
-};
-
-const makeSpaceAfter = (trx, order, SectionId) => {
-  return trx("Pages")
     .where({ SectionId })
-    .andWhere("order", ">", order)
-    .increment("order", SPACING);
+    .orderBy("order");
+
+  return isNil(id) ? query : query.where("id", "!=", id);
 };
 
-const movePage = async (trx, { id, sectionId, position }) => {
-  position = Math.max(0, position);
+const getMaxOrder = flow(last, getOr(0, "order"));
 
-  let rows = await getAdjacentRows(trx, id, sectionId, position);
+const makeSpaceForInsert = async (trx, sectionId, id, position) => {
```

Comment:
Not sure what to call this function. This doesn't seem like a great name, as it doesn't always "make space". Sometimes it's just doing a calculation, and others it's updating rows in the DB.

Anyone have any thoughts on this?

Reply:
Could (get/find)OrderForInsert() work?